app/controllers/games_controller.rb

```diff
@@ -1,24 +1,19 @@
 class GamesController < ApplicationController
   before_action :set_game, only: %i[ show edit update destroy most_winner ]
-  before_action :authenticate_user!, only: %i[user_games uesr_game new show edit create update destroy]
+  before_action :authenticate_user!, only: %i[new show edit create update destroy]
 
   # GET /games or /games.json
   def index
-    @games = Game.includes(:sessions).left_joins(:sessions).group(:id).order('COUNT(sessions.id) DESC, created_at').first(5)
-    #render json: @games.to_json(:include => {:sessions => {only: [:id, :date, :victor]}})
-    render json: @games.as_json(include: :sessions)
-  end
-
-  def user_games
-    initial = Game.joins(sessions: :user).left_joins(sessions: {session_shares: :collaborator})
-    @games = params[:name] ? Game.includes(:sessions).filter_by_name(params[:name]) : 
-    initial.where('session_shares.collaborator_id' => current_user.id).or(initial.where('sessions.user_id' => current_user.id))
-    .group(:id).order('COUNT(sessions.id) DESC, created_at')
-
+    if current_user
+      @games = user_games
+    else
+      @games = Game.includes(:sessions).left_joins(:sessions).group(:id).order('COUNT(sessions.id) DESC, created_at').first(5)  
```

Comment:
I'm not sure we need both the `includes` and the `left_joins` here. Do we get the same thing if we do the following?

```
def top_5_games
  Game
    .left_joins(:sessions)
    .select('games.*, COUNT(sessions.id) AS sessions_count')
    .group('games.id')
    .order('sessions_count DESC, games.created_at')
    .limit(5)
end
```

Also, this may be another good query candidate to put in a scope on the `Game` model. Maybe `Game.top_five`?

Reply:
I like the idea of adding this as another scope!